File: OOP_final_project_JAVA_version/src/exercicio_incremental_parte2/Aluno.java
package exercicio_incremental_parte2;

import java.util.Arrays;
import java.util.Date;

public class Aluno extends PessoaFisica{
    /*private String rua;
    private int numero;
    private String complemento;
    private String bairro;
    */
    private Endereco endereco;
    private String telefone;
    private int matricula;
    private String serie;
    private Escola escola;
    private PontoDeParada pontoDeParada;

    public Aluno(int matricula, String serie, String nome, String mae, String pai, String naturalidade, Date data_nascimento, String nome_oficial, String cpf_cnpj, Endereco endereco, String telefone) {
        super(nome, mae, pai, naturalidade, data_nascimento, nome_oficial, cpf_cnpj, endereco, telefone);
        this.matricula = matricula;
        this.serie = serie;
        this.endereco = endereco;
    }

    /*
    public Aluno(String rua, int numero, String complemento, String bairro, String telefone, int matricula, String serie) {
        this.rua = rua;
        this.numero = numero;
        this.complemento = complemento;
        this.bairro = bairro;
        this.telefone = telefone;
        this.matricula = matricula;
        this.serie = serie;
    }
    
    */
    
    
    @Override
    public void apresentarDados() {
        super.apresentarDados();
        System.out.println(endereco.toString());
        System.out.println("Matrícula: " + getMatricula());
        System.out.println("Série: " + getSerie());
    }
    
    @Override
    public String verificarTipo() {
        String tipoSuperClasse = super.verificarTipo();
        return tipoSuperClasse + " Aluno";
    }

    public int getMatricula() {
        return matricula;
    }

    public void setMatricula(int matricula) {
        this.matricula = matricula;
    }

    public String getSerie() {
        return serie;
    }

    public void setSerie(String serie) {
        this.serie = serie;
    }
    
    
}

File: OOP_final_project_python_version/aluno.py
from pessoa_fisica import PessoaFisica

class Aluno(PessoaFisica):
    def __init__(self, matricula, serie, nome, mae, pai, naturalidade, data_nascimento, nome_oficial, cpf_cnpj, endereco, telefone):
        super().__init__(nome, mae, pai, naturalidade, data_nascimento, nome_oficial, cpf_cnpj, endereco, telefone)
        self.__matricula = matricula
        self.__serie = serie
        

    def apresentar_dados(self):
        super().apresentar_dados()
        #print(self.endereco)
        print("Matrícula: " + str(self.matricula))
        print("Série: " + self.serie)

    def verificar_tipo(self):
        tipo_super_classe = super().verificar_tipo()
        return tipo_super_classe + " Aluno"

    @property
    def matricula(self):
        return self.__matricula

    @matricula.setter
    def matricula(self, matricula):
        self.__matricula = matricula

    @property
    def serie(self):
        return self.__serie

    @serie.setter
    def serie(self, serie):
        self.__serie = serie
